pkg/modulewriter/tfwriter.go

```diff
@@ -324,25 +293,19 @@ func writeProviders(vars map[string]cty.Value, dst string) error {
 		provBlock := hclBody.AppendNewBlock("provider", []string{prov})
 		provBody := provBlock.Body()
 		if _, ok := vars["project_id"]; ok {
-			pidToken := simpleTokenFromString("var.project_id")
-			pidTokens := []*hclwrite.Token{&pidToken}
-			provBody.SetAttributeRaw("project", pidTokens)
+			provBody.SetAttributeRaw("project", simpleTokens("var.project_id"))
 		}
 		if _, ok := vars["zone"]; ok {
-			zoneToken := simpleTokenFromString("var.zone")
-			zoneTokens := []*hclwrite.Token{&zoneToken}
-			provBody.SetAttributeRaw("zone", zoneTokens)
+			provBody.SetAttributeRaw("zone", simpleTokens("var.zone"))
 		}
 		if _, ok := vars["region"]; ok {
-			regToken := simpleTokenFromString("var.region")
-			regTokens := []*hclwrite.Token{&regToken}
-			provBody.SetAttributeRaw("region", regTokens)
+			provBody.SetAttributeRaw("region", simpleTokens("var.region"))
 		}
 		hclBody.AppendNewline()
 	}
 
 	// Write file
-	hclBytes := handleLiteralVariables(hclFile.Bytes())
```

Comment:
It appears to me that this func was not used in `packerwriter.go` and therefore no changes are needed in `packerwriter.go` (which only writes `.tfvars` attributes files)

Reply:
Correct, packerwrite only writes down result of `ResolveVariables` that shouldn't have raw HCL literals.